com.sap.cloud.lm.sl.cf.process/src/main/java/com/sap/cloud/lm/sl/cf/process/steps/DeleteServicesStep.java

```diff
@@ -83,23 +89,28 @@ protected String getStepErrorMessage(DelegateExecution context) {
         return Messages.ERROR_DELETING_SERVICES;
     }
 
-    private Map<String, CloudServiceExtended> getServicesData(List<String> serviceNames, ExecutionWrapper execution) {
+    private List<CloudServiceExtended> getServicesData(List<String> serviceNames, ExecutionWrapper execution) {
         CloudControllerClient client = execution.getControllerClient();
 
         return serviceNames.parallelStream()
-            .map(name -> client.getService(name, false))
-            .filter(Objects::nonNull)
-            .map(service -> ImmutableCloudServiceExtended.builder()
-                .metadata(service.getMetadata())
-                .name(service.getName())
-                .build())
-            .collect(Collectors.toMap(CloudEntity::getName, e -> e));
+                           .map(name -> client.getService(name, false))
+                           .filter(Objects::nonNull)
+                           .map(this::buildCloudServiceExtended)
+                           .collect(Collectors.toList());
+    }
+
+    private ImmutableCloudServiceExtended buildCloudServiceExtended(CloudService service) {
+        return ImmutableCloudServiceExtended.builder()
+                                            .metadata(service.getMetadata())
+                                            .name(service.getName())
+                                            .build();
     }
 
-    private List<String> getServicesWithoutData(List<String> servicesToDelete, Map<String, CloudServiceExtended> servicesData) {
-        return servicesToDelete.stream()
-            .filter(name -> servicesData.get(name) == null)
-            .collect(Collectors.toList());
+    private List<String> getServicesWithoutData(List<String> servicesToDelete, List<CloudServiceExtended> servicesData) {
+        List<String> servicesNames = servicesData.stream()
+                                                 .map(CloudServiceExtended::getName)
+                                                 .collect(Collectors.toList());
+        return new ArrayList<>(CollectionUtils.disjunction(servicesToDelete, servicesNames));

```

Comment:
Arrays.asList and also try to use Collection.removeAll or List::removeAll

Reply:
done